AppController/lib/taskqueue.rb

```diff
@@ -92,10 +106,43 @@ def self.start_master(clear_data, verbose)
   end
 
 
+  # Interacts with RabbitMQ, instructing to join the master's cluster, in
+  # different ways.
+  #
+  # Args
+  #   type: One of :join :update :reset_join, to indicate how to join the
+  #     cluster.
+  #   master_tq_host: The hostname of the master TaskQueue node.
+  # Returns
+  #   A Boolean indicating if all was good, of some failure happened.
+  def self.join_cluster(type, master_tq_host)
+    Djinn.log_info("Trying to #{type} #{master_tq_host} rabbitmq cluster.")
+    begin
+      Timeout.timeout(MAX_WAIT_FOR_RABBITMQ) do
+        Djinn.log_run("#{RABBIT_CTL} stop_app")
+        case type
+        when :join
+          Djinn.log_run("#{RABBIT_CTL} join_cluster rabbit@#{master_tq_host}")
+        when :update
```

Comment:
Yes. If the straight join_cluster fails, it could be because the node is already registered in the cluster. In this situation, the upate will pull in the latest data about this cluster, and re-sync the node.